docs/02-app/01-building-your-application/06-optimizing/02-videos.mdx

```diff
@@ -0,0 +1,281 @@
+
+For a comprehensive list of iframe attributes, refer to the MDN documentation [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attributes).
+
+### Choosing Between `<video>` and `<iframe>`
+
+Deciding whether to use `<video>` or `<iframe>` in your Next.js application depends on the source of your video content and your specific requirements. For self-hosted or direct video files where you need more control over the player's functionality and appearance, the `<video>` tag is ideal. On the other hand, for embedding content from platforms like YouTube or Vimeo, which offer their own set of features and functionalities, `<iframe>` is the preferred choice.
+
+### Displaying Externally Hosted Videos in Next.js
+
+Efficiently displaying videos from platforms like YouTube or Vimeo in Next.js involves combining appropriate embedding techniques with dynamic loading strategies. Using server components with a loading state managed by React Suspense optimizes page load times and enhances user experience. This ensures videos are loaded responsively, adapting to user interactions and network conditions.
+
+1. **Create a Server Component for Video Embedding**
+
+   The first step is to create a [Server Component](https://nextjs.org/docs/app/building-your-application/rendering/server-components) that generates the appropriate iframe for embedding the video. This component will fetch the source URL for the video and render the iframe.
+
+   ```jsx filename="app/ui/video-component.jsx"
+   export const async function VideoComponent(){
+     const src = await getVideoSrc()
+
+     return <iframe src={src} frameborder="0" allowfullscreen />
+   }
+   ```
+
+2. **Dynamically Load the Video Component Using React Suspense**
+
+   After creating the server component for video embedding, the next step is to [dynamically load](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming) this component. [React Suspense](https://react.dev/reference/react/Suspense) is used to manage the loading fallback, showing a fallback while the video is being fetched.
+
+   ```jsx filename="app/page.jsx"
+   import { Suspense } from 'react'
+   import { VideoComponent } from '../ui/VideoComponent.server'
+
+   export default function Page() {
+     return (
+       <section>
+         <Suspense fallback={<p>Loading video...</p>}>
+           <VideoComponent />
+         </Suspense>
+         {/* Other content of the page */}
+       </section>
+     )
+   }
+   ```
+
+> **Good to know**: When embedding videos from external platforms, consider the following best practices:
+>
+> - **Responsive Embedding**: Ensure the video embeds are responsive. Use CSS to make the iframe or video player adapt to different screen sizes.
+> - **Consider Data Usage**: Implement strategies for loading videos based on network conditions, especially for users with limited data plans.
```

Comment:
Is there a resource we can link to for "strategies for loading videos". Or would you say streaming with suspense is enough?